src/trajectory/util.cpp

```diff
@@ -315,5 +368,72 @@ UniqueSplinePtr createPartialTrajectory(
   return outputTrajectory;
 }
 
+//==============================================================================
+aikido::trajectory::ConstInterpolatedPtr toR1JointTrajectory(
+    ConstStateSpacePtr& space, const Interpolated& trajectory)
+{
+  checkValidityOfSpaceAndTrajectory(space, &trajectory);
+
+  auto interpolator = std::dynamic_pointer_cast<const GeodesicInterpolator>(
+      trajectory.getInterpolator());
+  if (!interpolator)
+    throw std::invalid_argument(
+        "The interpolator of trajectory should be a GeodesicInterpolator");
+
+  // Create new trajectory space.
+  std::vector<aikido::statespace::ConstStateSpacePtr> subspaces;
+  for (std::size_t i = 0; i < space->getDimension(); ++i)
+    subspaces.emplace_back(std::make_shared<const R1>());
+
+  auto rSpace = std::make_shared<CartesianProduct>(subspaces);
+  auto rInterpolator = std::make_shared<GeodesicInterpolator>(rSpace);
+  auto rTrajectory = std::make_shared<Interpolated>(rSpace, rInterpolator);
+
+  Eigen::VectorXd sourceVector(space->getDimension());
+  auto sourceState = rSpace->createState();
+
+  // Add the first waypoint
```

Comment:
I don't think this is meant to match the actual R1 value of the robot. It produces R1 trajectory such that two waypoints `[pi - alpha, -pi + alpha]` becomes `[pi - alpha, pi + alpha]` so that the smoothers(postprocessors) which assume R1 space can handle it gracefully. In the end, it is converted back to SO2 space before sending it via ROS, and low-level controllers are capable of handling it. @aditya-vk please correct me &/ confirm.

Reply:
That's right. This has nothing to do with the actual robot joint angle value. As @gilwoolee mentioned, this is only for timing to happen gracefully since the timers assume the values are R1.

Reply:
Got it, thanks for the clarification.